soundconverter/ui.py

```diff
@@ -276,20 +275,49 @@ def add_uris(self, uris, base=None, extensions=None):
         self.typefinders.queue_ended = self.typefinder_queue_ended
         self.typefinders.start()
 
+        self.window.set_status('{}'.format(_('Adding Files...')))
+
+        # show progress and enable gtk main loop iterations
+        # so that the ui stays responsive
+        self.window.progressbarstatus.set_text('0/{}'.format(len(files)))
+        self.window.progressbarstatus.set_show_text(True)
         while(self.typefinders.running):
+            if self.typefinders.progress:
+                completed = int(self.typefinders.progress * len(files))
+                self.window.progressbarstatus.set_fraction(self.typefinders.progress)
+                self.window.progressbarstatus.set_text('{}/{}'.format(completed, len(files)))
             gtk_iteration()
             time.sleep(0.1)
+        self.window.progressbarstatus.set_show_text(False)
+
         log('adding: %d files' % len(files))
+
+        # see if one of the files with an audio extension
+        # was not readable
+        known_audio_types = ['.flac', '.mp3', '.aac',
+            '.m4a', '.mpeg', '.opus', '.vorbis', '.ogg', '.wav']
+
+        broken_audiofiles = 0
         for f in files:
+            sound_file = SoundFile(f, base)
+            extension = os.path.splitext(f)[1].lower()
             if f not in self.good_files:
-                print('could not read', f)
+                if extension in known_audio_types:
+                    broken_audiofiles += 1
                 continue
-            sound_file = SoundFile(f, base)
             if sound_file.uri in self.filelist:
                 log('file already present: \'%s\'' % sound_file.uri)
                 continue
             self.append_file(sound_file)
 
+        if broken_audiofiles > 0:
+            s = ''
+            if broken_audiofiles > 1:
+                s = 's'
+            show_error('{} audiofile{} could not be read by gstreamer!'.format(broken_audiofiles, s),
```

Comment:
This will be very problematic to translate.

Reply:
You have to use ngettext() for this.

Reply:
fixed in fd173843bb14eb9ba7e92c7a0300e992723db731